metadata.go

```diff
@@ -9,7 +9,11 @@ func (r topicMetadataRequestV1) size() int32 {
 }
 
 func (r topicMetadataRequestV1) writeTo(w *bufio.Writer) {
-	writeStringArray(w, []string(r))
+	if len([]string(r)) > 0 {
```

Comment:
Heh heh...in some places, we get by without this check when serializing arrays b/c kafka usually don't differentiate between null and empty.

Reply:
interesting. gonna test without but it was failing for me.

Reply:
Oh, I believe that it was failing.  Just saying that Kafka is inconsistent on this front!